luwak/src/main/java/uk/co/flax/luwak/BatchQueryIndexConfiguration.java

```diff
@@ -0,0 +1,47 @@
+package uk.co.flax.luwak;
+
+/*
+ *   Copyright (c) 2015 Lemur Consulting Ltd.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+/**
+ * Encapsulates various configuration settings for a Monitor's query index
+ */
+public class BatchQueryIndexConfiguration {
+
+    private long commitTimeout = 0;
+
+    /**
+     * Set the commit Timeout (commitWithin)
+     *
+     * If set to 0 (default), then all queries are committed to the index immediately.
+     * If set to non-zero, then the commit is delayed by up to that many milliseconds.
+     * This means that multiple updates might be committed together.
+     *
+     * @param millis the timeout to use for commits
+     * @return the current configuration
+     */
+    public BatchQueryIndexConfiguration setCommitTimeout(long millis) {
+        this.commitTimeout = millis;
+        return this;
+    }
+
```

Comment:
True, I guess this originally assumed that we we inherit from the QueryIndexConfiguration object.  But point taken, this can go away and commitTimeout can be an argument to BatchMonitor instead.